### DISCUSSION TITLE
Program stuck on calling `YMConnect::CloseConnection`

### DISCUSSION THREAD

**Discussion by Gray-Stone** [Yaskawa-Global/YMConnect#23] [Q&A] [answered]
I made an executable following the examples in `MotionManagerInterface` document. With the main addition of another thread requesting robot position.

At the end of my executable. I called `YMConnect::CloseConnection` with the `MotomanController` pointer. However, the executable just stuck at this line and never returned. I have stopped the other thread that's requesting position, stop the move, clear trajectory, and turn off servo. But nothing helped.

On the other hand, if I never try to move the robot, then I won't get this issue.

From most of the scripts I see, `YMConnect::OpenConnection` is called at start of the program. However I didn't see an example using `YMConnect::CloseConnection` .

Are there anything specific I need to do before calling CloseConnection for it to finish properly?

**Comment by aseamans** [comment]
Hi @Gray-Stone 
The CloseConnection mainly calls a destructor for MotomanController object and there isn't anything that you need to do before you call it. Does this occur if you are running the example in the docs without the addition of the thread requesting robot position?

**Reply by Gray-Stone** [reply]
Same thing happened if I turn off the joint value reading thread. 

Is the library designed to be safe for multi threading access?

**Reply by aseamans** [reply]
It is. I will try to replicate your code when I come in tomorrow and see if I can get the same error and track down what's going on. Are you using the newest version of the library?

**Reply by Gray-Stone** [reply]
I think it should be the latest. This is what's on top of the header.
`#define YMConnect_Header_Version "1.0.6"`

**Reply by aseamans** [reply]
Are you running Windows or Linux?

**Reply by Gray-Stone** [reply]
Linux Ubuntu 22.04

**Reply by aseamans** [reply, marked as answer]
[YMConnect.zip](https://github.com/user-attachments/files/17234594/YMConnect.zip)
Try using this new header file and lib. Let me know if it works and I will push out a new release. Thanks for letting us know about this.

**Reply by Gray-Stone** [reply]
This new version does fix the issue. Thanks.
